src/components/BlogPost/BlogPostDialog.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { FC } from 'react';
+
+import BlogPostDialogMobile from './BlogPostDialogMobile';
+import BlogPostDialogDesktop from './BlogPostDialogDesktop';
+import BlogPostDialogTablet from './BlogPostDialogTablet';
+
+export type BlogPostDialogProps = {
+  title: string;
+  subtitle: string;
+  tagtitle: string;
+  contentheader: string;
+  imgurl: string;
+  contentmain: string;
+  isOpen: boolean;
+};
+
+type BlogPostDialogType = 'mobile' | 'tablet' | 'desktop';
+
+export type DetailsBlogPostDialogType = {
+  type: BlogPostDialogType;
+};
+export const BlogPostDialog: FC<BlogPostDialogProps & DetailsBlogPostDialogType> = ({ isOpen, type, ...props }) => {
```

Comment:
BlogPostDialogType is unnecessary
DetailsBlogPostDialogType can be extended(&) BlogPostDialogProps by type: 'mobile' | 'tablet' | 'desktop';  This type doesn't need to be exported
Now we can write: FC<DetailsBlogPostDialogType>